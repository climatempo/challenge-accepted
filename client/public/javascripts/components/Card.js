class Card {

    template() {
        return `<card class="card col-6">
            <div class="card-content">
                <div class="date-card">${this.date}</div>
                <div class="message-card">${this.text}</div>
                <div class="infos-card">
                    <div class="block-card">
                        ${this.max}
                        ${this.min}
                    </div>
                    <div class="block-card">
                        ${this.precipitation}
                        ${this.probability}
                    </div>
                </div>
            </div>
        </card>`;
    }

    build() {
        const {
            date,
            temperature: {
                max,
                min
            },
            rain: {
                precipitation,
                probability
            }
        } = this;

        this.date = new Date(date).toLocaleDateString();

        const objmax = new MaxTemperature(max);
        this.max = objmax.template();

        const objmin = new MinTemperature(min);
        this.min = objmin.template();

        const objprecipitation = new Precipitation(precipitation);
        this.precipitation = objprecipitation.template();

        const objprobability = new Probability(probability);
        this.probability = objprobability.template();

        return this;
    }
}