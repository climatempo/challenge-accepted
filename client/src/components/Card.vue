<template>
  <div class="card col-6">
    <div class="card-content">
      <div class="date-card">{{ card.date | formatDate }}</div>
      <div class="message-card">{{ card.text }}</div>
      <div class="infos-card">
        <div class="block-card">
          <div class="max-temp cell-card">
            <i class="img-max image"></i>
            <p class="temperature-max temperature">{{ card.temperature.max }}</p>
          </div>
          <div class="min-temp cell-card">
            <i class="img-min image"></i>
            <p class="temperature-min temperature">{{ card.temperature.min }}</p>
          </div>
        </div>
        <div class="block-card">
          <div class="precipitation cell-card">
            <i class="img-precipitation image"></i>
            <p class="text-precipitation temperature">{{ card.rain.precipitation }}</p>
          </div>
          <div class="probability cell-card">
            <i class="img-probability image"></i>
            <p class="text-probability temperature">{{ card.rain.probability }}</p>
          </div>
        </div>
      </div>
    </div>
  </div>
</template>

<script>
export default {
  name: 'card',
  props: ['card'],
  filters: {
    formatDate(date) {
      return date
        .split('-')
        .reverse()
        .join('/');
    },
  },
};
</script>

<style scoped>
</style>
