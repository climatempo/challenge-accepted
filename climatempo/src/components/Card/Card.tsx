import React from "react";
import styled from "styled-components";
import max from "../../assets/icons/max.png";
import min from "../../assets/icons/min.png";
import probabilityIcon from "../../assets/icons/probability.png";
import precipitationIcon from "../../assets/icons/precipitation.png";

type TypeCard = {
  date: string;
  text: string;
  temperaturaMin: number;
  temperaturaMax: number;
  probability: number;
  precipitation: number;
};

function Card({
  date,
  text,
  temperaturaMin,
  temperaturaMax,
  probability,
  precipitation,
}: TypeCard) {
  return (
    <Container>
      <Header>
        <Date>{date}</Date>
        <Description>{text}</Description>
      </Header>
      <Body>
        <Column>
          <Row>
            <Icon src={max} alt="max" />
            <Row>
              <Number>{temperaturaMax}</Number>
              <Number>°C</Number>
            </Row>
          </Row>
          <Row>
            <Icon src={probabilityIcon} alt="probability" />
            <Row>
              <Number>{precipitation}</Number>
              <Number>mm</Number>
            </Row>
          </Row>
        </Column>
        <Column>
          <Row>
            <Icon src={min} alt="min" />
            <Row>
              <Number>{temperaturaMin}</Number>
              <Number>°C</Number>
            </Row>
          </Row>
          <Row>
            <Icon src={precipitationIcon} alt="precipitation" />
            <Row>
              <Number>{probability}</Number>
              <Number>%</Number>
            </Row>
          </Row>
        </Column>
      </Body>
    </Container>
  );
}

const Container = styled.div`
  width: 346px;
  height: 230px;
  background: #d9d9d9;
  border-radius: 20px;
  display: flex;
  flex-direction: column;

  @media (max-width: 888px) {
    width: 346px;
    height: 230px;
  }

  @media (max-width: 468px) {
    width: 274px;
    height: 220px;
  }
`;

const Header = styled.div`
  width: 346px;
  height: 80px;
  background: #047ab2;
  border-radius: 20px;
  display: flex;
  flex-direction: column;
  align-items: center;

  @media (max-width: 888px) {
    width: 346px;
    height: 80px;
  }

  @media (max-width: 468px) {
    width: 274px;
    height: 160px;
  }
`;

const Date = styled.div`
  margin-top: 10px;
  margin-left: 10px;
  font-family: "Roboto", sans-serif;
  font-style: normal;
  font-weight: 400;
  font-size: 18px;
  line-height: 24px;
  color: #ffffff;

  @media (max-width: 888px) {
    font-size: 18px;
  }

  @media (max-width: 468px) {
    font-size: 16px;
  }
`;

const Description = styled.p`
  width: 80%;
  margin-bottom: 10px;
  margin-left: 10px;
  font-family: "Roboto", sans-serif;
  font-style: normal;
  font-weight: 400;
  font-size: 13px;
  line-height: 14px;
  color: #ffffff;
`;

const Body = styled.div`
  display: flex;
  flex-direction: row;
`;

const Column = styled.div`
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  width: 50%;
`;

const Icon = styled.img`
  width: 25px;
  height: 25px;

  @media (max-width: 868px) {
    width: 20px;
    height: 20px;
  }

  @media (max-width: 468px) {
    width: 20px;
    height: 20px;
  }
`;

const Row = styled.div`
  width: 80px;
  display: flex;
  flex-direction: row;
  justify-content: center;
  align-items: center;
`;

const Number = styled.p`
  margin-left: 10px;
  font-family: "Roboto", sans-serif;
  font-style: normal;
  font-weight: 400;
  font-size: 24px;
  line-height: 29px;
  color: #000000;

  @media (max-width: 868px) {
    font-size: 18px;
  }

  @media (max-width: 468px) {
    font-size: 18px;
  }
`;

export default Card;
