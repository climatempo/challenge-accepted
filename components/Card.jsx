import style from "styles/components/Card.module.scss"

function Card(props) {
  return <div className={style.card}>Card</div>
}

export default Card
