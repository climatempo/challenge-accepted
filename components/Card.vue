<template lang="html">
  <b-card
    no-body
    style="width: 100%"
  >
    <template v-slot:header>
      <h6>{{ date }}</h6>
      <p>{{ text }}</p>
    </template>

    <b-card-body class="">
      <div class="row">
        <div class="col-6">
          <img src="/images/icons/upload.png" alt="">
          <span class="text-primary">{{ max }}</span>
        </div>
        <div class="col-6">
          <img src="/images/icons/download.png" alt="">
          <span class="text-danger">{{ min }}</span>
        </div>
        <div class="col-6 pt-3">
          <img src="/images/icons/raindrop-close-up.png" alt="">
          <span class="text-dark">{{ water }}</span>
        </div>
        <div class="col-6 pt-3">
          <img src="/images/icons/protection-symbol-of-opened-umbrella-silhouette-under-raindrops.png" alt="">
          <span class="text-dark">{{ rain }}</span>
        </div>
      </div>
    </b-card-body>
  </b-card>
</template>

<script>
export default {
}
</script>

<style lang="css" scoped>
.card-body {
  padding: 10px;
}
.row {
  margin: 0px;
}
.col-6 {
  display: flex;
  align-items: center;
  padding: 3px;
  padding-top: 0px;
}
.col-6 span {
  width: 100%;
  text-align: center;
  font-size: 22px;
}
p {
  font-size: 12px;
  margin-bottom: 0px;
}
h6 {
  font-weight: bold;
}
.text-primary {
  color: #127fc0;
}
.text-danger {
  color: #ca3e3e;
}
.text-dark {
  color: #000;
}
</style>
