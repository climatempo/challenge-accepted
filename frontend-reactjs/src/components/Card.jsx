import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';

const Container = styled.div`
  box-shadow: 0 5px 5px 1px rgba(0, 0, 0, .1);
  min-height: 200px;
  background-color: #eee;
  margin: 10px 0;
`;

const Card = ({ children }) => (
  <Container>{children}</Container>
);

Card.propTypes = {
  children: PropTypes.element,
};

export default Card;
