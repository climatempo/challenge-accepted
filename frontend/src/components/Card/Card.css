/*============= CONFIG MOBILE =============*/
@media screen and (max-width: 500px){
    .Card {
        width: 80%;
        height: 11rem;
        background-color: #e9e9e9;
        border-radius: 25px;
        display: flex;
        flex-direction: column;
        box-shadow: 2px 2px rgb(168, 168, 168);
        min-width: 300px; 
        margin-bottom: 1rem;
    }
    
    .Content {
        display: flex;
        background-color: rgb(243, 241, 241);
        flex: 2;
        align-items: center;
        border-radius: 25px 25px 0px 0px ;
        padding: 0.4rem;
    }
    
    .Date {
        display: flex;
        flex: 1;
        padding: 0.8rem;
        font-size: 0.9rem;
        font-weight: bold;
        margin-left: 0.3rem;
        margin-right: -1rem;
    }
    
    .Data {
        display: flex;
        flex: 3;
        justify-content: center;
        font-weight: bold;
    }
    
    .TxtData{
        font-size: 0.9rem;
        line-height: 2.3rem;
    }
    
    .Icon {
        width: 1.3rem;
        padding: 0rem 0.5rem ;
        margin-left: 0.5rem;
    }

    .Degree1{
        color: rgb(228, 42, 42);
    }

    .Degree2{
        color: rgb(42, 123, 228);
    }
    
    .Content2 {
        display: flex;
        flex:1;
        text-align: start;
        padding: 1rem;
        font-size: 0.8rem;
        background-color: #e3e3e3;
        border-radius: 0px 0px 25px 25px;
        padding: 1.2rem;
        word-spacing: 1px;
        line-height: 0.9rem;
    }
}

/*============= CONFIG WEB =============*/

@media screen and (min-width:500px) {
    .Card {
        width: 90%;
        height: 7rem;
        background-color: #e9e9e9;
        border-radius: 25px;
        display: flex;
        flex-direction: row;
        box-shadow: 2px 2px rgb(168, 168, 168);
        min-width: 650px;
        max-width: 900px;
        margin-bottom: 1rem;
    }
    
    .Content {
        display: flex;
        background-color: rgb(243, 241, 241);
        flex: 1;
        align-items: center;
        border-radius: 25px 0px 0px 25px;
        min-width: 40%;
        max-width: 60%;
        font-size: 0.95rem;
        padding: 0.4rem;
    }
    
    .Date {
        display: flex;
        flex: 1;
        padding: 0.8rem;
        font-size: 0.9rem;
        font-weight: bold;
        margin-left: 0.5rem;
    }
    
    .Data {
        display: flex;
        flex: 3;
        justify-content: space-between;
        font-weight: bold;
    }
    
    .TxtData {
        padding: 1rem;
        font-size: 0.9;
        line-height: 2.5rem;
    }
    
    .Icon {
        width: 1.3rem;
        padding: 0 0.3rem;
        margin-left: 0.5rem;
    }

    .Degree1 {
        color: rgb(228, 42, 42);
    }

    .Degree2 {
        color: rgb(42, 123, 228);
    }
    
    .Content2 {
        display: flex;
        flex:1;
        text-align: start;
        padding: 1rem;
        font-size: 0.9rem;
        line-height: 1.2rem;
        background-color: #e3e3e3;
        border-radius: 0px 25px 25px 0px;
        max-width: 60%;
        align-items: center;
        padding: 1.2rem;
    }
}

