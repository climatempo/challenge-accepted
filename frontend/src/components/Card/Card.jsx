import React from 'react'
import "./Card.css";
import arrowdown from "../images/icons/setadown.png";
import arrowup from "../images/icons/setaup.png";
import drop from "../images/icons/raindrop.png";
import umbrella from "../images/icons/umbrella.png";



export default function Card(props) {
    return (
        <div className='Card'>
            <div className='Content'>
                <p className='Date'>{props.date}</p>
                <div className='Data'>
                    <table>
                        <tbody>
                            <tr className='TxtData'>
                                <td><img className='Icon' src={arrowup} alt="" /><span className='Degree1'>{props.temMax}°{props.temUnit}</span></td>
                                <td><img className='Icon' src={arrowdown} alt="" /><span className='Degree2'>{props.temMin}°{props.temUnit}</span></td>
                            </tr>
                            <tr className='TxtData'>
                                <td><img className='Icon' src={drop} alt="" /><span>{Number(props.precipitation).toFixed(1)}{props.precUnit}</span></td>
                                <td><img className='Icon' src={umbrella} alt="" /><span>{props.probability}%</span></td>
                            </tr>                           
                        </tbody>
                    </table>
                </div>
            </div>
            <div className='Content2'>
                <p>{props.txt}</p>
            </div>
        </div>
    )
    
};