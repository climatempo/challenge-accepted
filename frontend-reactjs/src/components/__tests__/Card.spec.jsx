import React from 'react';
import toJson from 'enzyme-to-json';

import { mount } from '../../tests-setup';

import Card from '../Card';

describe('Card Component', () => {
  let wrapper = null;

  beforeEach(() => {
    wrapper = mount(
    <Card><h1>foo</h1></Card>,
    );
  });

  it('should correctly render', () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('Should exists element h1', () => {
    expect(wrapper.find('h1').exists()).toBeTruthy();
  });

  it('Should must contain the text "foo"', () => {
    expect(wrapper.find('h1').text()).toBe('foo');
  });
});
