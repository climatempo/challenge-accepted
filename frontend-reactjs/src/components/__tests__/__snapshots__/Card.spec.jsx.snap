// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card Component should correctly render 1`] = `
<Card>
  <styled.div>
    <StyledComponent
      forwardedComponent={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "sc-bdVaJa",
            "isStatic": true,
            "lastClassName": "eUbZKV",
            "rules": Array [
              "
  box-shadow: 0 5px 5px 1px rgba(0, 0, 0, .1);
  min-height: 200px;
  background-color: #eee;
  margin: 10px 0;
",
            ],
          },
          "displayName": "styled.div",
          "foldedComponentIds": Array [],
          "render": [Function],
          "styledComponentId": "sc-bdVaJa",
          "target": "div",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        }
      }
      forwardedRef={null}
    >
      <div
        className="sc-bdVaJa eUbZKV"
      >
        <h1>
          foo
        </h1>
      </div>
    </StyledComponent>
  </styled.div>
</Card>
`;
